### PR TITLE
feat(connector): [BANKOFAMERICA] Add Connector Template Code

### PR DESCRIPTION
**Pull request by deepanshu-iiitu**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Add BankOfAmerica template code.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

BankOfAmerica connector integration
VAS PR: https://github.com/juspay/hyperswitch-cloud/pull/3061

## How did you test it?

Template code added hence no testing required

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/api_models/src/enums.rs [hunk 1]
```diff
@@ -195,6 +196,7 @@ pub enum RoutableConnectors {
     Adyen,
     Airwallex,
     Authorizedotnet,
+    // Bankofamerica, Added as template code for future usage
     Bitpay,
     Bambora,
```
Comment (prasunna09) [inline crates/api_models/src/enums.rs:201]: Can you please put them in the alphabetical order?

File: crates/router/tests/connectors/sample_auth.toml [hunk 2]
```diff
@@ -183,4 +183,9 @@ api_key="API Key"
 api_key="API Key"
 
 [prophetpay]
-api_key="API Key"
\ No newline at end of file
+api_key="API Key"
+
+[bankofamerica]
+api_key = "Bearer MyApiKey"
```
Comment (prasunna09) [inline crates/router/tests/connectors/sample_auth.toml:189]: Can you please remove this bearer, we will be concatenating `bearer` in connector.rs?

**Review by prasunna09 — CHANGES_REQUESTED** [review]

File: crates/api_models/src/enums.rs (hunk 1 above)
Comment (deepanshu-iiitu) [reply crates/api_models/src/enums.rs:201]: Done

File: crates/router/tests/connectors/sample_auth.toml (hunk 2 above)
Comment (deepanshu-iiitu) [reply crates/router/tests/connectors/sample_auth.toml]: Done

**Review by prasunna09 — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**